### ISSUE TITLE
Setting duration does not work as it should

### ISSUE THREAD

**Issue by IAkumaI** [zerodevx/svelte-toast#89]
```js
const id = toast.push('message', {duration: 90000});
toast.set(id, {msg: 'another', duration: 3000});
```

Hello. The code above must change the durations of the toast, but it does not.
As a "fix" - you can hover the toast and the duration will change after and only after the hovering.

Svelte. The latest version.

**Comment by zerodevx** [comment]
Thanks for the report - it's a drawback of the current implementation. What is your expectation though? Does the second statement cause the progress bar to reset (move from 1 to 0) in 3s?

**Comment by IAkumaI** [comment]
The expectation was changing of the duration :)
I was showing a toast "Wait..." then I set it to "Success/Failure" and didn't know how to do it right.

Current (and correct) version is:
```js
const id = toast.push('message', {dismissable: false, initial: 0, next: 0, duration: 3000});
toast.set(id, {msg: 'another', dismissable: true, next: 1});
```

But the bug is still there, so I decided to report it.
The second statement does nothing with the duration. The progress bar didn't react at all until I hovered the toasts. I suppose there are some updates on-hover and it causes the update.

**Comment by zerodevx** [comment]
Currently `duration` is not reactive - I'll fix it in the next release.

**Comment by Spimy** [comment]
Are there any plans for this to be fixed / added soon? It has been a while since this issue was opened.

**Comment by zerodevx** [comment]
Yes, cutting a `v1` release is next on my to-do list, once I finish this round of work.
